client/src/App.js

```diff
@@ -201,9 +203,11 @@ export default class App extends Component {
 
   async onThreadClickHandler(threadId) {
     this.setState({ composingNewThread: false });
+    this.setState({ currentFromNumber: utils.threadIdToFromNumber(threadId) });
     await this.controller.getMessagesForThread(threadId);
     await this.controller.refreshThreads();
     this.textInput.focus();
+    console.log(this.state.currentFromNumber);
```

Comment:
Just a note here that `setState` is asynchronous so the updated value above may not be accurately represented here. It might be worth doing something like this:
```
const currentFromNumberToSet = utils.threadIdToFromNumber(threadId);
this.setState({ currentFromNumber: currentFromNumberToSet });
console.log(`currentFromNumberToSet: ${currentFromNumberToSet}`);